File: src/market-tracker-webapi/Application/Http/Models/Schemas/List/ShoppingListOutputModel.cs
using market_tracker_webapi.Application.Domain.Schemas.List;

namespace market_tracker_webapi.Application.Http.Models.Schemas.List;

public record ShoppingListOutputModel(
    string Id,
    string Name,
    DateTime? ArchivedAt,
    DateTime CreatedAt,
    Guid OwnerId,
    bool IsOwner,
    bool IsArchived,
    int NumberOfMembers
);

public static class ListOutputModelMapper
{
    public static ShoppingListOutputModel ToOutputModel(this ShoppingList list, Guid authUserId)
    {
        return new ShoppingListOutputModel(
            list.Id.Value,
            list.Name,
            list.ArchivedAt,
            list.CreatedAt,
            list.OwnerId.Value,
            list.OwnerId.Value == authUserId,
            list.IsArchived,
            list.MemberIds.Count()
        );
    }
}

File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/http/models/list/ShoppingListOutputModel.kt
package pt.isel.markettracker.http.models.list

import java.time.LocalDateTime

data class ShoppingListOutputModel(
    val id: String,
    val name: String,
    val archivedAt: LocalDateTime?,
    val createdAt: LocalDateTime,
    val ownerId: String,
    val isOwner: Boolean,
    val isArchived: Boolean
)